app/lib/sanitize_config.rb

```diff
@@ -19,19 +19,45 @@ module Config
       node['class'] = class_list.join(' ')
     end
 
+    UNSUPPORTED_HREF_TRANSFORMER = lambda do |env|
+      return unless env[:node_name] == 'a'
+
+      current_node = env[:node]
+
+      scheme = begin
+        if current_node['href'].blank?
+          nil
+        else
+          uri = Addressable::URI.parse(current_node['href'])
+
+          if uri.relative?
+            :relative
+          else
+            uri.scheme
+          end
+        end
+      rescue Addressable::URI::InvalidURIError
+        nil
+      end
```

Comment:
This whole thing is done without fully parsing URLs in Sanitize
cf. https://github.com/rgrove/sanitize/blob/master/lib/sanitize/transformers/clean_element.rb#L143

It's probably safer (in terms of possible changed behaviors) and more CPU-efficient to do it the same way.

Btw, I think we should either reject URLs or rewrite them to be relative to the original status (an option I prefer, but is maybe not worth it).